src/python/pants/backend/scala/compile/scalac.py

```diff
@@ -184,33 +191,43 @@ async def compile_scala_source(
                 scala_artifacts.compiler_main,
                 "-bootclasspath",
                 ":".join(tool_classpath.classpath_entries(toolcp_relpath)),
-                *local_plugins.args(local_scalac_plugins_relpath),
+                *scalac_plugins.args(scalac_plugins_relpath),
                 *(("-classpath", classpath_arg) if classpath_arg else ()),
                 *scalac.args,
-                # NB: We set a non-existent main-class so that using `-d` produces a `jar` manifest
-                # with stable content.
-                "-Xmain-class",
-                NO_MAIN_CLASS,
                 "-d",
-                output_file,
+                output_dir,
```

Comment:
When using `scalac` plugins that emit additional files (like `semanticdb` or `scalajs` do), `scalac` will not produce a `jar` anymore (even when using the `-Xmain-class` argument) and it will instead output all compiled binaries into a directory.